cachito/workers/pkg_managers/yarn.py

```diff
@@ -69,7 +69,22 @@ def get_yarn_proxy_repo_username(request_id):
     return f"cachito-yarn-{request_id}"
 
 
-def _get_deps(yarn_lock, file_deps_allowlist):
+def find_non_dev(non_dev_deps, yarn_lock, current_dep):
+    if current_dep not in non_dev_deps:
+        non_dev_deps.append(current_dep)
+        sub_deps = []
+        for dep_identifier, dep_data in yarn_lock.items():
+            if current_dep in dep_identifier:
+                if "dependencies" in dep_data:
+                    sub_deps = [
+                        f"{name}@{version}" for name, version in dep_data["dependencies"].items()
+                    ]
+                break
+        for sub_dep in sub_deps:
+            find_non_dev(non_dev_deps, yarn_lock, sub_dep)
```

Comment:
I stopped reading here, replace recursion by stack and iteration.

Reply:
Got it!